### PR TITLE
default rr

### PR DESCRIPTION
**Pull request by mejango**
# Description

currently, if reserved token beneficiary isn't set, reserved tokens are not mintable. Instead, don't let rr to be set in the first place if no beneficiary is set.

## Limitations & risks

*Are there any trade-off or new vulnarbility surface based on theses changes?*

# Check-list
- [x] Tests are covering the new feature
- [x] Code is [natspec'd](https://docs.soliditylang.org/en/v0.8.17/natspec-format.html)
- [x] Code is [linted and formatted](https://docs.soliditylang.org/en/v0.8.17/style-guide.html)
- [x] I have run the test locally (and they pass)
- [x] I have rebased to the latest main commit (and tests still pass)

# Interactions
These changes will impact the following contracts:

- Directly:
JBTiered721DelegateStore.sol
- Indirectly:

### CODE REVIEW

File: contracts/JBTiered721DelegateStore.sol [hunk 1]
```diff
@@ -685,9 +698,12 @@ contract JBTiered721DelegateStore is IJBTiered721DelegateStore {
       if (_flags.lockVotingUnitChanges && _tierToAdd.votingUnits != 0)
         revert VOTING_UNITS_NOT_ALLOWED();
 
-      // Make sure a reserved rate isn't set if changes should be locked or if manual minting is allowed.
+      // Make sure a reserved rate isn't set if changes should be locked, if manual minting is allowed, or if a reserved rate beneficiary isn't set.
       if (
-        (_flags.lockReservedTokenChanges || _tierToAdd.allowManualMint) &&
+        (_flags.lockReservedTokenChanges ||
+          _tierToAdd.allowManualMint ||
+          (_tierToAdd.reservedTokenBeneficiary == address(0) &&
```
Comment (mejango) [inline contracts/JBTiered721DelegateStore.sol:705]: i think no need for this anymore. will check in a sec.

File: contracts/forge-test/NFTReward_Unit.t.sol [hunk 2]
```diff
@@ -1790,6 +1790,87 @@ contract TestJBTieredNFTRewardDelegate is Test {
...
+        })
+      );
+      _delegate.test_store().ForTest_setReservesMintedFor(
+        address(_delegate),
+        i + 1,
+        reservedMinted
+      );
+    }
+
+    for (uint256 i; i < 10; i++)
```
Comment (mejango) [inline contracts/forge-test/NFTReward_Unit.t.sol:1867]: @viraj124 please also make sure when the tier is viewed in the `teirs` getters, they return with reservedRate of 0 despite being originally set with a non-zero rr.

Comment (viraj124) [reply contracts/forge-test/NFTReward_Unit.t.sol:1882]: should be good now

File: contracts/JBTiered721DelegateStore.sol [hunk 3]
```diff
@@ -812,13 +825,13 @@ contract JBTiered721DelegateStore is IJBTiered721DelegateStore {
 
       // Set the reserved token beneficiary if needed.
       if (_tierToAdd.reservedTokenBeneficiary != address(0))
-        if (_tierToAdd.shouldUseReservedTokenBeneficiaryAsDefault)
+        if (_tierToAdd.shouldUseReservedTokenBeneficiaryAsDefault && defaultReservedTokenBeneficiaryOf[msg.sender] != _tierToAdd.reservedTokenBeneficiary)
           defaultReservedTokenBeneficiaryOf[msg.sender] = _tierToAdd.reservedTokenBeneficiary;
```
Comment (drgorillamd) [inline contracts/JBTiered721DelegateStore.sol:829]: defaultReservedTokenBeneficiaryOf is read from storage twice, if the stack allows, would be better to load it once

File: contracts/JBTiered721DelegateStore.sol [hunk 4]
```diff
@@ -812,13 +825,13 @@ contract JBTiered721DelegateStore is IJBTiered721DelegateStore {
...
         else _reservedTokenBeneficiaryOf[msg.sender][_tierId] = _tierToAdd.reservedTokenBeneficiary;
```
Comment (drgorillamd) [inline contracts/JBTiered721DelegateStore.sol:830]: This isn't tested @viraj124  and got a doubt:

if `_tierToAdd.shouldUseReservedTokenBeneficiaryAsDefault` is true but `defaultReservedTokenBeneficiaryOf[msg.sender] == _tierToAdd.reservedTokenBeneficiary`
-> defaultReservedBenef is already set, so good, but we end up setting  _reservedTokenBeneficiaryOf[msg.sender][_tierId] too (iic, defautReserved isn't read anymore then, even if we change it later on - Ln#687)

Same bellow for royalty

Comment (mejango) [reply contracts/JBTiered721DelegateStore.sol:830]: nice catch

**Pull request merged**